src/Application/ApprovalLog.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\PageApprovals\Application;
+
+interface ApprovalLog {
+
+	public function getApprovalState( int $pageId ): ?ApprovalState;
+
+	public function unapprovePage( int $pageId, int $userId ): void;
+
+	public function approvePage( int $pageId, ?int $userId ): void;
```

Comment:
Shouldn't the unapprovePage userId be the nullable one? In the case where a user edits a page and causes it to be unapproved, didn't we say that's when the userId becomes empty? (i.e. the current user didn't manually unapprove, and should not be logged)
Whereas when a page is approved, there is a always a manual approver, which is the current user.

Reply:
I guess the comment in ApprovalState confirms the above for automatic unapproval: https://github.com/ProfessionalWiki/PageApprovals/pull/12/files#diff-a28bda54c5854c66bb5c35881799e2034732131317b6bad6ee95667ee4fec93eR10